### PR TITLE
sqlite: complete Pod methods

### PR DESCRIPTION
**Pull request by vrothberg**
@mheon @baude PTAL

This PRs completes the Pod methods and adds a number of fixes.   I was chasing the failure in `test/system/220-healthcheck.bats` but ran out of time.  Many system and e2e tests are passing, so it's now in a state where we can play on the CLI.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by vrothberg** [comment]
Note that I'll be buried in meetings until the end of the day.  Feel free to take over this PR if needed.

File: libpod/sqlite_state.go [hunk 1]
```diff
@@ -64,15 +64,15 @@ func NewSqliteState(runtime *Runtime) (_ State, defErr error) {
...
-	if err := state.migrateSchemaIfNecessary(); err != nil {
-		return nil, err
-	}
-
 	// Set up tables
 	if err := sqliteInitTables(state.conn); err != nil {
 		return nil, fmt.Errorf("creating tables: %w", err)
 	}
 
+	if err := state.migrateSchemaIfNecessary(); err != nil {
```
Comment (vrothberg) [inline libpod/sqlite_state.go:72]: @mheon there's probably a better way but this change unblocked me.

File: libpod/sqlite_state.go [hunk 2]
```diff
@@ -1160,38 +1159,41 @@ func (s *SQLiteState) RewritePodConfig(pod *Pod, newCfg *PodConfig) error {
...
+	if err != nil {
+		return fmt.Errorf("updating pod config table with new configuration for pod %s: %w", pod.ID(), err)
+	}
+	rows, err := results.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("retrieving pod %s config rewrite rows affected: %w", pod.ID(), err)
+	}
+	if rows == 0 {
+		pod.valid = false
+		return define.ErrNoSuchPod
```
Comment (baude) [inline libpod/sqlite_state.go:1189]: should this be errorf and cite the pod?

File: libpod/sqlite_state.go [hunk 3]
```diff
@@ -1340,47 +1315,22 @@ func (s *SQLiteState) HasPod(id string) (bool, error) {
...
-	// 	}
-
-	// 	return nil
-	// })
-	// if err != nil {
-	// 	return false, err
-	// }
+	var check int
+	if err := row.Scan(&check); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
```
Comment (baude) [inline libpod/sqlite_state.go:1322]: we seem to do this a lot in the DB code (including the container code). do we need something like IsEmpty|NoRows ?

File: libpod/sqlite_state.go [hunk 4]
```diff
@@ -1816,119 +1573,36 @@ func (s *SQLiteState) RemovePodContainers(pod *Pod) error {
...
+		if err := rows.Scan(&id); err != nil {
+			return fmt.Errorf("scanning container from database: %w", err)
+		}
 
-	// 	return nil
-	// })
-	// if err != nil {
-	// 	return err
-	// }
+		if err := s.removeContainerWithTx(id, tx); err != nil {
```
Comment (baude) [inline libpod/sqlite_state.go:1600]: should this just be return func() ?

Comment (baude) [reply libpod/sqlite_state.go:1600]: ah i see, its in a loop, nvm

**Comment by baude** [comment]
code lgtm, couple of nits

File: libpod/sqlite_state.go (hunk 1 above)
Comment (mheon) [reply libpod/sqlite_state.go:72]: Hm. It needs to happen before table init to be safe, so we'll probably have to figure out how to ignore the table-does-not-exist error.

Comment (mheon) [reply libpod/sqlite_state.go:72]: Can do that later though

File: libpod/sqlite_state.go (hunk 3 above)
Comment (mheon) [reply libpod/sqlite_state.go:1322]: Wouldn't save us much in terms of code... Go is super verbose around errors, no fixing that.

**Comment by vrothberg** [comment]
@mheon feel free to merge and take over the comments.  I won't find time today.

**Pull request merged**